block-storage/src/main/scala/coop/rchain/blockstorage/BlockDagFileStorage.scala

```diff
@@ -381,115 +431,135 @@ final class BlockDagFileStorage[F[_]: Concurrent: Sync: Log: BlockStore] private
+                                  //they were bonded (i.e. this block)
+                                  case (acc, v) => acc.updated(v, block.blockHash)
+                                }
+                              }
+                            }
+                        _ <- updateLatestMessagesFile(
+                              (newValidators + block.sender).toList,
+                              block.blockHash
+                            )
+                        _ <- updateDataLookupFile(blockMetadata)
+                      } yield newValidatorsWithSenderEither.map(_ => ())
                     }
-                _ <- updateLatestMessagesFile(
-                      (newValidators + block.sender).toList,
-                      block.blockHash
-                    )
-                _ <- updateDataLookupFile(blockMetadata)
-              } yield ()
-            }
         dag <- representation
-      } yield dag
+      } yield insertion.map(_ => dag)
     )
 
-  def checkpoint(): F[Unit] =
-    ().pure[F]
+  def checkpoint(): F[StorageErr[Unit]] =
+    ().asRight[StorageError].pure[F]
 
-  def clear(): F[Unit] =
+  def clear(): F[StorageErr[Unit]] =
     lock.withPermit(
-      for {
-        latestMessagesLogOutputStream <- getLatestMessagesLogOutputStream
-        _                             <- Sync[F].delay { latestMessagesLogOutputStream.close() }
-        blockMetadataLogOutputStream  <- getBlockMetadataLogOutputStream
-        _                             <- Sync[F].delay { blockMetadataLogOutputStream.close() }
-        _ <- Sync[F].delay {
-              Files.write(latestMessagesDataFilePath, Array.emptyByteArray)
-            }
-        _ <- Sync[F].delay {
-              Files.write(blockMetadataLogPath, Array.emptyByteArray)
-            }
-        newLatestMessagesCrc      = Crc32.empty[F]()
-        newLatestMessagesCrcBytes <- newLatestMessagesCrc.bytes
-        _ <- Sync[F].delay {
-              Files.write(latestMessagesCrcFilePath, newLatestMessagesCrcBytes)
-            }
-        newBlockMetadataCrc      = Crc32.empty[F]()
-        newBlockMetadataCrcBytes <- newBlockMetadataCrc.bytes
-        _ <- Sync[F].delay {
-              Files.write(blockMetadataCrcPath, newBlockMetadataCrcBytes)
-            }
-        _ <- setDataLookup(Map.empty)
-        _ <- setChildMap(Map.empty)
-        _ <- setTopoSort(Vector.empty)
-        _ <- setLatestMessages(Map.empty)
-        _ <- setLatestMessagesLogOutputStream(
-              new FileOutputStream(latestMessagesDataFilePath.toFile)
+      (for {
+        latestMessagesLogOutputStream <- EitherT.liftF[F, StorageError, FileOutputStreamIO[F]](
+                                          getLatestMessagesLogOutputStream
+                                        )
+        _ <- toStorageErrT(latestMessagesLogOutputStream.close)
+        blockMetadataLogOutputStream <- EitherT.liftF[F, StorageError, FileOutputStreamIO[F]](
```

Comment:
I'm having a hard time going through all those lifts and other changes, and I think we shouldn't maybe even bother passing the storage error **for handling** upstream (bare with me :smile:). It wasn't available for handling upstream in the first place (until this PR), so the only handling possible should be 'kill the app' / whatever our top level handlers do.

I'm starting to think we should require a `FunctorRaise[F, StorageError]` constraint in the file io wrappers, provide an instance for `Sync[F] / MonadError[F, Throwable]`, make the wrappers return `F[A]` (with the FunctorRaise constraint) instead of `F[StorageIOErr[A]]`, and have little-to-no changes in this file in this commit. WDYT?

Reply:
I'm skipping this commit for now